packages/ember-data/lib/system/references/belongs-to.js

```diff
@@ -0,0 +1,77 @@
+import Ember from 'ember';
+// FIXME doesn't work an Model is undefined
+// import Model from '../model';
+import Reference from './reference';
+
+var BelongsToReference = function(store, parentInternalModel, belongsToRelationship) {
+  this._super$constructor(store, parentInternalModel);
+  this.belongsToRelationship = belongsToRelationship;
+  this.type = belongsToRelationship.relationshipMeta.type;
+
+  if (belongsToRelationship.link) {
+    this.remoteType = "link";
+    this.link = belongsToRelationship.link;
+  } else {
+    this.remoteType = "id";
+    this.id = belongsToRelationship.inverseRecord && belongsToRelationship.inverseRecord.id;
+  }
+
+  // TODO parent
+  // TODO inverse
+};
+
+BelongsToReference.prototype = Ember.create(Reference.prototype);
+BelongsToReference.prototype.constructor = BelongsToReference;
+BelongsToReference.prototype._super$constructor = Reference;
+
+BelongsToReference.prototype.meta = function() {
+  return this.parent.meta;
+};
+
+BelongsToReference.prototype.push = function(objectOrPromise) {
+  return Ember.RSVP.resolve(objectOrPromise).then((data) => {
+    var record;
+
+    // FIXME getting trolled by `import Model from '../model' since it is
+    // undefined :woot: maybe because the references are not included in
+    // main.js??
+    // TODO add check for polymorphic case
+    if (data instanceof DS.Model) {
+      record = data;
+    } else {
+      record = this.store.push(this.type, data);
+    }
+
+    this.belongsToRelationship.setCanonicalRecord(record._internalModel);
+
+    return record;
+  });
+};
+
+BelongsToReference.prototype.value = function() {
+  var inverseRecord = this.belongsToRelationship.inverseRecord;
+  return inverseRecord && inverseRecord.record;
+};
+
+BelongsToReference.prototype.load = function() {
+  if (this.remoteType === "id") {
+    return this.store.find(this.type, this.id);
+  }
+
+  if (this.remoteType === "link") {
+    return this.belongsToRelationship.findLink().then((internalModel) => {
+      return this.value();
+    });
+  }
+};
+
+BelongsToReference.prototype.reload = function() {
+  var record = this.value();
```

Comment:
No, becasue this code is incorrect, so it's easier to make it correct by adding reload onto relationship than hacking here